### PR TITLE
refactor: remove old learner record from Credentials

### PR DESCRIPTION
**Pull request by jsnwesson**
### Context

The Learner's Record view has been extracted into the Learner Record MFE, and so the record view that exists in Credentials can be safely removed. 

[[DEPR]: Program Record UI #1722](https://github.com/openedx/credentials/issues/1722
)

### What this removes

- Program Record view
- Record List view
- Jest configurations and dependencies
- React configurations and dependencies
- Eslint configurations and dependencies

**Run JavaScript tests locally with Karma**

There is work being done on a fix to get Karma to run in CI. Until then, however, contributors are required to run these tests locally.

- [ ] Make sure you are inside the devstack container
- [ ] Run `make test-karma`
- [ ] All tests pass

### CODE REVIEW

**Comment by justinhynes** [comment]
The PR is basically what I expected but I noticed that there are some CI issues. I didn't want to approve this without seeing a clean test run.

File: package.json [hunk 1]
```diff
@@ -6,8 +6,6 @@
     "url": "git://github.com/openedx/credentials"
   },
   "scripts": {
-    "test-react": "jest --no-cache --coverage --config ./jest-config.json",
-    "lint": "eslint --ext .js --ext .jsx ./credentials/static"
```
Comment (justinhynes) [inline package.json:10]: If the Makefile references these NPM scripts, could we ensure to update the Makefile as well? Thanks!

Comment (justinhynes) [reply package.json:10]: Nevermind, I see you already handled this. This is the problem with looking at a PR over multiple days haha

Comment (jsnwesson) [reply package.json:10]: haha! No worries, thankfully our CI was able to catch this, otherwise I wouldn't have known!

**Comment by jsnwesson** [comment]
closing this PR to break it down into smaller chunks 
https://github.com/openedx/credentials/pull/2104
https://github.com/openedx/credentials/pull/2106

**Pull request closed without merging**